File: api/src/main/java/be/gilmotech/nestspend/dto/futureevent/ProjectionResponse.java
package be.gilmotech.nestspend.dto.futureevent;

import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDate;
import java.util.List;

/**
 * Response DTO for budget projections.
 */
@Schema(description = "Budget projection response with projected balances over time")
public record ProjectionResponse(
        @Schema(description = "Start date of the projection period")
        LocalDate startDate,

        @Schema(description = "End date of the projection period")
        LocalDate endDate,

        @Schema(description = "Initial balance in cents at projection start")
        Long initialBalanceCents,

        @Schema(description = "Projected balance in cents at projection end")
        Long finalBalanceCents,

        @Schema(description = "Monthly projection data points")
        List<ProjectionDataPoint> dataPoints
) {
}

File: web/nestspend-web/src/app/core/api/models/projection-response.ts
/* eslint-disable */

import { ProjectionDataPoint } from './projection-data-point';

/**
 * Budget projection response with projected balances over time
 */
export interface ProjectionResponse {
  /**
   * Start date of the projection period
   */
  startDate?: string;

  /**
   * End date of the projection period
   */
  endDate?: string;

  /**
   * Initial balance in cents at projection start
   */
  initialBalanceCents?: number;

  /**
   * Projected balance in cents at projection end
   */
  finalBalanceCents?: number;

  /**
   * Monthly projection data points
   */
  dataPoints?: Array<ProjectionDataPoint>;
}
